docker/entrypoint.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -e
+
+# following recommendations for a consistent interface:
+# https://github.com/docker-library/official-images#consistency
+
+# this if will check if the first argument is a flag
+# but only works if all arguments require a hyphenated flag
+# -v; -SL; -f arg; etc will work, but not arg1 arg2
+if [ "${1:0:1}" = '-' ]; then
+    set -- simp_le "$@"
+fi
+
+# check for the expected command
+if [ "$1" = 'simp_le' ]; then
+    shift
+    exec simp_le "$@"
+fi
+
+# else default to run whatever the user wanted like "bash"
+exec "$@"
```

Comment:
This could be a bit simpler:

```bash
case "$1" in
	-*) exec simp_le $@ ;;
	'') exec simp_le ;;
	*) exec "$@"  ;;
esac
```

Reply:
```
case "$1" in
	-*) exec simp_le "$@" ;;
	simp_le) shift; exec simp_le "$@" ;;
	*) exec "$@"  ;;
esac
```

Works as intended. The idea is to be able to run the container pretty much like you run the command (after the docker specific part), but remain able to bypass the default cmd of the container (`/usr/bin/simp_le --help` in this case):

`docker run zenhack/simp_le` will run simp_le with the --help argument.

`docker run zenhack/simp_le -v` and `docker run zenhack/simp_le simp_le -v` will both run simp_le with the -v argument.

`docker run -it zenhack/simp_le bash` will run an interactive bash session inside the container.

This is a prerequisite if you want to be able to base other containers on this one.

Reply:
Yep, the interface is fine, I was mostly quibbling about the implementation, which I thought could be simpler.

In your version of the case statement, the second line seems superfluous. If it weren't there you'd just fall through to the one below it, which seems like it should do the same thing (rather than shifting away 'simp_le' and then adding it again, you just leave it there in the first place). Am I missing something?

Reply:
You're not, the second line is indeed superfluous, I'll remove it.